packages/cubejs-schema-compiler/src/compiler/CubeSymbols.ts

```diff
@@ -292,7 +292,8 @@ export class CubeSymbols {
         // If the hierarchy is included all members from it should be included as well
         // Extend `includes` with members from hierarchies that should be auto-included
         const cubes = type === 'dimensions' ? cube.cubes.map((it) => {
-          const fullPath = this.evaluateReferences(null, it.joinPath, { collectJoinHints: true });
+          // TODO recheck `it.joinPath` typing
+          const fullPath = this.evaluateReferences(null, it.joinPath as () => { toString(): string }, { collectJoinHints: true });
```

Comment:
Maybe it would be better to define an interface for `() => { toString(): string }` instead of copy-pasting many times it `as () => { toString(): string }` ? At least for now. wdyt?